docs: add documentation and for default git hook containing DCO and c… (#482)
aws-deadline/deadline-cloud: CONTRIBUTING.md

[GitHub documentation on the topic](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message)
to guide you.

### Digital Certificate of Origin.

This repository enforces the Developer Certificate of Origin (DCO) on Pull Requests. It requires all commit messages to contain the `Signed-off-by` line with an email address that matches the commit author.

> To generate the `Signed-off-by`, make sure `git commit` is passed `-s` option.

Reviewer: ```suggestion
To generate the `Signed-off-by` line, make sure `git commit` is passed with the `-s` flag.
```

Author: Great suggestion! I'll change it.

### Automating Conventional Commits and DCO.

To simplify Conventional Commit and DCO, it is recommended to setup a Git hook for the repository. Add the following as a git hook:

- Edit `{repository}/.git/hooks/commit-msg`

> - `chmox +x {repository}/.git/hooks/commit-msg`

Reviewer: ```suggestion
- `chmod +x {repository}/.git/hooks/commit-msg`
```

Author: Darn, I thought I commited Jericho's suggestion.

> - `chmox +x `{repository}/.git/hooks/commit-msg`

Reviewer 2: ```suggestion
- `chmod +x {repository}/.git/hooks/commit-msg`
```

I'm assuming this is supposed to be chmod? Also an extra backtick in there

Author: Good one! Thanks!
